modules/@apostrophecms/image/ui/apos/components/AposMediaManager.vue

```diff
@@ -440,31 +462,103 @@ export default {
       this.selectAll();
       this.editing = undefined;
     },
-    async updatePage(num) {
-      if (num) {
-        this.currentPage = num;
-        await this.getMedia();
-      }
-    },
     archiveClick() {
       this.$emit('archive', this.checked);
     },
 
-    async search(query) {
-      this.filter('autocomplete', query);
+    async search(value) {
+      this.filterValues.autocomplete = value;
+      this.currentPage = 1;
+      this.items = [];
+      this.isLoading = true;
+      await this.debouncedGetMedia();
+      this.isLoading = false;
     },
 
-    async onContentChanged() {
-      await this.getMedia({ tags: true });
+    async onContentChanged({ action, doc }) {
+      if (doc.type !== '@apostrophecms/image' || ![ 'archive', 'update' ].includes(action)) {
+        return;
+      }
+      if (this.modified || action === 'archive') {
+        await this.refetchMedia({ tags: true });
+        return;
+      }
+
+      await this.updateEditing(null);
+    },
+
+    async handleIntersect(entries) {
+      for (const entry of entries) {
+        if (
+          entry.isIntersecting &&
+          this.currentPage < this.totalPages &&
+          !this.isFirstLoading &&
+          this.items.length
+        ) {
+          this.currentPage++;
+          this.isScrollLoading = true;
+          await this.$nextTick();
+          this.loadRef.scrollIntoView({
```

Comment:
since loadRef is not always defined, can this be dangerous?

Reply:
I added a guard, this observer is created only if a ref is passed:
```javascript
    setLoadRef(ref) {
      if (ref) {
        this.loadRef = ref;
        this.observeLoadRef();
      }
    }

```